ethers/providers/jsonrpc.nim

```diff
@@ -25,10 +25,15 @@ type
   JsonRpcSigner* = ref object of Signer
     provider: JsonRpcProvider
     address: ?Address
-  JsonRpcProviderError* = object of EthersError
+  JsonRpcProviderError* = object of ProviderError
   SubscriptionHandler = proc(id, arguments: JsonNode): Future[void] {.gcsafe, upraises:[].}
 
-template raiseProviderError(message: string) =
+proc raiseProviderError(message: string) {.upraises: [JsonRpcProviderError].} =
+  var message = message
+  try:
+    message = parseJson(message){"message"}.getStr
+  except Exception:
```

Comment:
Yes, I'd like that, but the `push: {.upraises: [].}` forces me to handle all exceptions that the compiler thinks this code can throw, and that unfortunately includes `Exception`.

Reply:
Fair enough, it would be great if `parseJson` didn't raise `Exception`. We could use [`json_serialization`](https://github.com/status-im/nim-json-serialization) to avoid the `except Exception`, but that would be a bit overkill for this situation.